components/odh-notebook-controller/controllers/notebook_runtime_test.go

```diff
@@ -3,25 +3,132 @@ package controllers
 import (
 	"context"
 	"fmt"
+	"time"
 
-	nbv1 "github.com/kubeflow/kubeflow/components/notebook-controller/api/v1"
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
+
+	nbv1 "github.com/kubeflow/kubeflow/components/notebook-controller/api/v1"
+	imagev1 "github.com/openshift/api/image/v1"
 	corev1 "k8s.io/api/core/v1"
 	apierrs "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-var _ = Describe("When Creating a notebook should mount the configMap", func() {
+const (
+	resource_reconciliation_timeout      = time.Second * 30
+	resource_reconciliation_check_period = time.Second * 2
+	Namespace                            = "default"
+	RuntimeImagesCMName                  = "pipeline-runtime-images"
+	expectedMountName                    = "runtime-images"
+	expectedMountPath                    = "/opt/app-root/pipeline-runtimes/"
+)
+
+var _ = Describe("Runtime images ConfigMap should be mounted", func() {
 	ctx := context.Background()
 
-	When("Creating a Notebook", func() {
+	When("Empty ConfigMap for runtime images", func() {
+
+		BeforeEach(func() {
+			err := cli.Create(ctx, &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: odhNotebookControllerTestNamespace}}, &client.CreateOptions{})
+			if err != nil && !apierrs.IsAlreadyExists(err) {
+				Expect(err).ToNot(HaveOccurred())
+			}
+		})
+
+		testCases := []struct {
+			name         string
+			notebook     *nbv1.Notebook
+			notebookName string
+			ConfigMap    *corev1.ConfigMap
+		}{
+			{
+				name:         "ConfigMap without data",
+				notebookName: "test-notebook-runtime-empty-cf",
+				ConfigMap: &corev1.ConfigMap{
+					TypeMeta: metav1.TypeMeta{
+						Kind:       "ConfigMap",
+						APIVersion: "v1",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      RuntimeImagesCMName,
+						Namespace: Namespace,
+					},
+					Data: map[string]string{},
+				},
+			},
+		}
+
+		for _, testCase := range testCases {
+			Context(fmt.Sprintf("The Notebook runtime pipeline images ConfigMap test case: %s", testCase.name), func() {
+				notebook := createNotebook(testCase.notebookName, Namespace)
+				configMap := &corev1.ConfigMap{}
+				It(fmt.Sprintf("Should mount ConfigMap correctly: %s", testCase.name), func() {
+
+					// cleanup first
+					_ = cli.Delete(ctx, notebook, &client.DeleteOptions{})
+					_ = cli.Delete(ctx, testCase.ConfigMap, &client.DeleteOptions{})
+
+					// wait until deleted
+					By("Waiting for the Notebook to be deleted")
```

Comment:
the `By` serves by way of the comment; why have both a comment and a By that says exactly the same thing?